c/meson.build

```diff
@@ -0,0 +1,106 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+project(
+    'arrow-adbc',
+    'c', 'cpp',
+    version: '0.6.0-SNAPSHOT',  # TODO: can this be dynamic?
+    license: 'Apache 2.0',
+    meson_version: '>=1.3.0',
+    default_options: [
+        'buildtype=release',
+        'c_std=gnu99',  # TODO: gmtime_r is gnu99 - likely unintentional std
+        'warning_level=2',
+        'cpp_std=c++17',
+    ]
+)
+
+root_dir = include_directories('..')
+driver_dir = include_directories('driver')
+nanoarrow_dep = dependency('nanoarrow')
+fmt_dep = dependency('fmt')
+
+if get_option('tests')
+    gtest_main_dep = dependency('gtest_main')
+    # Seems to be some kind of bug with meson test
+    # gtest-port.h:1770 pthread_key_delete(key_)failed with error 22
+    # https://github.com/google/googletest/pull/4536 ???
+    gtest_dep = dependency('gtest')
+    gmock_dep = dependency('gmock')
+else
+    gtest_dep = disabler()
+    gmock_dep = disabler()
+endif
+
+if get_option('duckdb')
```

Comment:
This might be tough to get working in its current state - probably will defer to fully fixing for a later PR

Reply:
Can we drop support for DukcDB with Meson?
I think that we don't need to maintain this because it's just for an integration test. Users don't need this and we can do it with CMake.